finagle-http/src/main/scala/com/twitter/finagle/http/codec/HttpServerDispatcher.scala

```diff
@@ -31,6 +31,11 @@ private[finagle] class HttpServerDispatcher(
     service.close()
   }
 
+  private def mayHaveContent(status: Status): Boolean = status match {
```

Comment:
I would definately wayt for @mosesn to chime in before going down the netty pipeline route. As a preview, I'll give my understanding of it. The code at https://github.com/twitter/finagle/blob/develop/finagle-http/src/main/scala/com/twitter/finagle/http/Codec.scala#L307 gives an example of how the `RespondToExpectContinue` handler works, but it is an `UpstreamHandler` (I believe that corresponds to `InboundHandler` in netty4). The `DownstreamHandler` works essentially the same but in the other direction. In my minds eye, the `DownstreamHandler` intercepts the `HttpResponse` and drops the body and requisite headers.

One thing that occurs to me is that streaming responses (chunked) don't appear to be addressed in this commit. The way streaming responses are rendered appears more complex and might make using the netty pipeline more tricky, if even possible. For example, In the case of a 304 or 204 I doubt that we want to use the `Reader`, and instead we would want to discard it. I don't think that is a possibility, at least in the place I pointed you to above.